spring-data-cassandra/src/main/java/org/springframework/data/cassandra/convert/AbstractCassandraConverter.java

```diff
@@ -41,18 +44,40 @@ public AbstractCassandraConverter(ConversionService conversionService) {
 
 	/**
 	 * Registers {@link EntityInstantiators} to customize entity instantiation.
-	 * 
+	 *
 	 * @param instantiators
 	 */
 	public void setInstantiators(EntityInstantiators instantiators) {
 		this.instantiators = instantiators == null ? new EntityInstantiators() : instantiators;
 	}
 
+	/**
+	 * Registers the given custom conversions with the converter.
+	 *
+	 * @param conversions
+	 */
+	public void setCustomConversions(CustomConversions conversions) {
+		this.conversions = conversions;
+	}
+
+	/**
+	 * Registers additional converters that will be available when using the {@link ConversionService} directly (e.g. for
+	 * id conversion). These converters are not custom conversions as they'd introduce unwanted conversions.
+	 */
+	private void initializeConverters() {
```

Comment:
Perhaps colocate `initializeConverters()` and move below `afterPropertiesSet()`.